ui/src/app/kvm/views/KVMList/KVMListTable/KVMListTable.js

```diff
@@ -0,0 +1,90 @@
+import { Col, MainTable, Row } from "@canonical/react-components";
+import React, { useEffect } from "react";
+import { useDispatch, useSelector } from "react-redux";
+
+import {
+  pod as podActions,
+  resourcepool as poolActions,
+  zone as zoneActions,
+} from "app/base/actions";
+import { pod as podSelectors } from "app/base/selectors";
+import CPUColumn from "./CPUColumn";
+import NameColumn from "./NameColumn";
+import OSColumn from "./OSColumn";
+import PoolColumn from "./PoolColumn";
+import PowerColumn from "./PowerColumn";
+import RAMColumn from "./RAMColumn";
+import StorageColumn from "./StorageColumn";
+import TypeColumn from "./TypeColumn";
+import VMsColumn from "./VMsColumn";
+
+const generateRows = (pods) => {
+  return pods.map((pod) => ({
+    key: pod.id,
+    columns: [
+      { content: <NameColumn id={pod.id} /> },
+      { content: <PowerColumn id={pod.id} /> },
+      { content: <TypeColumn id={pod.id} /> },
+      { className: "u-align--right", content: <VMsColumn id={pod.id} /> },
+      { content: <OSColumn id={pod.id} /> },
+      { content: <PoolColumn id={pod.id} /> },
+      { content: <CPUColumn id={pod.id} /> },
+      { content: <RAMColumn id={pod.id} /> },
+      { content: <StorageColumn id={pod.id} /> },
+    ],
+  }));
+};
+
+const KVMListTable = () => {
+  const dispatch = useDispatch();
+  const pods = useSelector(podSelectors.all);
+
+  useEffect(() => {
+    dispatch(podActions.fetch());
+    dispatch(poolActions.fetch());
+    dispatch(zoneActions.fetch());
+  }, [dispatch]);
+
+  return (
+    <Row>
+      <Col size={12}>
+        <MainTable
+          headers={[
+            { content: "FQDN" },
+            { content: "Power" },
+            { content: "KVM Host Type" },
+            {
+              className: "u-align--right",
+              content: (
+                <>
+                  <span>
+                    VM<span style={{ textTransform: "none" }}>s</span>
```

Comment:
Should this maybe be a utility class?

Reply:
@huwshimi Yeah probably. I'll update it now